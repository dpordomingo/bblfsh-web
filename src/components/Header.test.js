import React from 'react';
import { shallow } from 'enzyme';
import 'jest-styled-components';
import { Header, DriverCode } from './Header';
import renderer from 'react-test-renderer';

const testLanguages = {
  '': { name: 'auto' },
  python: { name: 'Python', url: 'Python.driver' },
  java: { name: 'Java', url: 'Java.driver' },
};

const testExamples = {
  python: {
    code: '',
  },
  java: {
    code: '',
  },
};

describe('DriverCode', () => {
  it('contains the driver URL if selectedLanguage is not auto', () => {
    const component = renderer.create(
      <DriverCode
        languages={testLanguages}
        actualLanguage="java"
        selectedLanguage="java"
        examples={testExamples}
      />
    );

    expect(component.toJSON()).toMatchStyledComponentsSnapshot();
  });

  it('contains the actualLanguage URL if is auto', () => {
    const component = renderer.create(
      <DriverCode
        languages={testLanguages}
        actualLanguage="python"
        selectedLanguage=""
        examples={testExamples}
      />
    );

    expect(component.toJSON()).toMatchStyledComponentsSnapshot();
  });
});

describe('Header', () => {
  it('renders correctly', () => {
    const component = renderer.create(
      <Header
        languages={testLanguages}
        actualLanguage="java"
        selectedLanguage=""
        examples={testExamples}
      />
    );

    expect(component.toJSON()).toMatchStyledComponentsSnapshot();
  });

  it('calls onLanguageChanged when the dropdown is changed', () => {
    const onLanguageChangedSpy = jest.fn();
    const wrapper = shallow(
      <Header
        languages={testLanguages}
        onLanguageChanged={onLanguageChangedSpy}
        actualLanguage="java"
        selectedLanguage=""
        examples={testExamples}
      />
    );

    wrapper.find('#language-selector').simulate('change', {
      target: { value: 'python' },
    });
    expect(onLanguageChangedSpy.mock.calls.length).toBe(1);
    expect(onLanguageChangedSpy.mock.calls[0][0]).toBe('python');
  });

  it('calls onExampleChanged when the dropdown is changed', () => {
    const spy = jest.fn();
    const wrapper = shallow(
      <Header
        languages={testLanguages}
        onExampleChanged={spy}
        actualLanguage="java"
        selectedLanguage=""
        selectedExample="python"
        examples={testExamples}
      />
    );

    wrapper.find('#examples-selector').simulate('change', {
      target: { value: 'java' },
    });
    expect(spy.mock.calls.length).toBe(1);
    expect(spy.mock.calls[0][0]).toBe('java');
  });

  it('calls onRunParser when the button is clicked', () => {
    const onRunParserSpy = jest.fn();
    const wrapper = shallow(
      <Header
        languages={testLanguages}
        onRunParser={onRunParserSpy}
        actualLanguage="java"
        selectedLanguage=""
        examples={testExamples}
      />
    );

    wrapper.find('#run-parser').simulate('click');
    expect(onRunParserSpy.mock.calls.length).toBe(1);
  });

  it('has the button disabled if canParse is false', () => {
    const component = renderer.create(
      <Header
        languages={testLanguages}
        canParse={false}
        actualLanguage="python"
        selectedLanguage=""
        examples={testExamples}
      />
    );

    expect(component.toJSON()).toMatchStyledComponentsSnapshot();
  });

  it('has the button enabled if canParse is true', () => {
    const component = renderer.create(
      <Header
        languages={testLanguages}
        canParse={true}
        actualLanguage="python"
        selectedLanguage=""
        examples={testExamples}
      />
    );

    expect(component.toJSON()).toMatchStyledComponentsSnapshot();
  });
});
