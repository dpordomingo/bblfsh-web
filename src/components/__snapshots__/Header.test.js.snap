// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DriverCode contains the actualLanguage URL if is auto 1`] = `
.elnEoP {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-left: 0.5rem;
}

.cANOcV {
  width: 20px;
  opacity: 0.7;
  margin-right: 0.3rem;
}

.ebxRNn {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: black;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 0.9rem;
}

.ebxRNn:hover {
  color: #EB7422;
}

.ebxRNn:hover .sc-VigVT {
  opacity: 1;
}

<div
  className="sc-gqjmRU elnEoP"
>
  <a
    className="sc-jTzLTM ebxRNn"
    href="Python.driver"
    target="_blank"
  >
    <img
      alt="Driver code repository on GitHub"
      className="sc-VigVT cANOcV"
      src="github.svg"
      title="Driver code repository on GitHub"
    />
  </a>
</div>
`;

exports[`DriverCode contains the driver URL if selectedLanguage is not auto 1`] = `
.elnEoP {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-left: 0.5rem;
}

.cANOcV {
  width: 20px;
  opacity: 0.7;
  margin-right: 0.3rem;
}

.ebxRNn {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: black;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 0.9rem;
}

.ebxRNn:hover {
  color: #EB7422;
}

.ebxRNn:hover .sc-VigVT {
  opacity: 1;
}

<div
  className="sc-gqjmRU elnEoP"
>
  <a
    className="sc-jTzLTM ebxRNn"
    href="Java.driver"
    target="_blank"
  >
    <img
      alt="Driver code repository on GitHub"
      className="sc-VigVT cANOcV"
      src="github.svg"
      title="Driver code repository on GitHub"
    />
  </a>
</div>
`;

exports[`Header has the button disabled if canParse is false 1`] = `
.dfgKLf {
  height: 70px;
  padding: 0 1rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: #f9f7f5;
  border-bottom: 1px solid #dad7d0;
  z-index: 9999;
  box-shadow: 0 5px 25px rgba(138, 128, 115, 0.14);
}

.gRdHVw {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-size: 1.5rem;
  font-weight: normal;
  padding: 0;
  margin: 0;
  height: 100%;
  border-right: 1px solid #dad7d0;
  padding-right: 1rem;
}

.iZWaVz {
  height: 40px;
}

.dtdqzi {
  margin-left: 0.8rem;
}

.gkhDQm {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-left: 1rem;
  height: 100%;
}

.dWtenm {
  color: grey;
  margin-right: 1em;
  font-size: 0.8rem;
  font-weight: bold;
  -webkit-text-transform: uppercase;
  text-transform: uppercase;
  color: #636262;
}

.eJfGpj {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 100%;
  border-right: 1px solid #dad7d0;
  padding-right: 1rem;
}

.eJfGpj + .eJfGpj {
  margin-left: 1rem;
}

.eJfGpj:last-child {
  border-right: none;
}

.jdktxT {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 100%;
  border-right: 1px solid #dad7d0;
  padding-right: 1rem;
  -webkit-flex-grow: 1;
  -ms-flex-grow: 1;
  flex-grow: 1;
  -webkit-flex-direction: row-reverse;
  -ms-flex-direction: row-reverse;
  flex-direction: row-reverse;
  padding-right: 0;
}

.jdktxT + .jdktxT {
  margin-left: 1rem;
}

.jdktxT:last-child {
  border-right: none;
}

.fvfSyI {
  border-radius: 3px;
  border: 1px solid #dad7d0;
  background: white;
  padding: 0.5em 0.5em;
  -webkit-text-transform: uppercase;
  text-transform: uppercase;
  font-weight: bold;
  font-size: 0.8rem;
  font-size: .7rem;
  -webkit-text-transform: uppercase;
  text-transform: uppercase;
}

.fvfSyI + button {
  margin-left: 3px;
}

.acdud {
  border-radius: 3px;
  border: 1px solid #dad7d0;
  background: white;
  padding: 0.5em 0.5em;
  -webkit-text-transform: uppercase;
  text-transform: uppercase;
  font-weight: bold;
  font-size: 0.8rem;
  -webkit-text-transform: none;
  text-transform: none;
}

.acdud + button {
  margin-left: 3px;
}

.kUlzXo {
  padding: 0.5em 0.5em;
  border: 1px solid #B1B1B1;
  border-radius: 3px;
  cursor: pointer;
  background: #B1B1B1;
  color: #212121;
  font-weight: bold;
  -webkit-text-transform: uppercase;
  text-transform: uppercase;
  font-size: 0.8rem;
  -webkit-letter-spacing: 0.05em;
  -moz-letter-spacing: 0.05em;
  -ms-letter-spacing: 0.05em;
  letter-spacing: 0.05em;
  -webkit-transition: box-shadow 300ms ease-in-out;
  transition: box-shadow 300ms ease-in-out;
  border: none;
}

.kUlzXo:hover {
  box-shadow: 0 5px 15px rgba(138, 128, 115, 0.8);
}

.kUlzXo + .kUlzXo {
  margin-left: 0.7rem;
}

.kmPTEb {
  padding: 0.5em 0.5em;
  border: 1px solid #B1B1B1;
  border-radius: 3px;
  cursor: pointer;
  background: #B1B1B1;
  color: #212121;
  font-weight: bold;
  -webkit-text-transform: uppercase;
  text-transform: uppercase;
  font-size: 0.8rem;
  -webkit-letter-spacing: 0.05em;
  -moz-letter-spacing: 0.05em;
  -ms-letter-spacing: 0.05em;
  letter-spacing: 0.05em;
  -webkit-transition: box-shadow 300ms ease-in-out;
  transition: box-shadow 300ms ease-in-out;
  padding: .7rem 1.8rem;
  background: #EB7422;
  color: #FFFFFF;
  white-space: nowrap;
}

.kmPTEb:hover {
  box-shadow: 0 5px 15px rgba(138, 128, 115, 0.8);
}

.kmPTEb:hover {
  box-shadow: 0 5px 15px rgba(191, 76, 50, 0.48);
}

.elnEoP {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-left: 0.5rem;
}

.cANOcV {
  width: 20px;
  opacity: 0.7;
  margin-right: 0.3rem;
}

.ebxRNn {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: black;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 0.9rem;
}

.ebxRNn:hover {
  color: #EB7422;
}

.ebxRNn:hover .sc-VigVT {
  opacity: 1;
}

<header
  className="sc-bdVaJa dfgKLf"
>
  <h1
    className="sc-bwzfXH gRdHVw"
  >
    <img
      alt="bblfsh"
      className="sc-htpNat iZWaVz"
      src="babelfish_logo.svg"
    />
    <span
      className="sc-bxivhb dtdqzi"
    >
      Dashboard
    </span>
  </h1>
  <div
    className="sc-ifAKCX gkhDQm"
  >
    <div
      className="sc-bZQynM eJfGpj"
    >
      <label
        className="sc-EHOje dWtenm"
        htmlFor="language-selector"
      >
        Language
      </label>
      <select
        className="sc-htoDjs fvfSyI"
        id="language-selector"
        onChange={[Function]}
        value=""
      >
        <option
          value=""
        >
          Python (auto)
        </option>
        <option
          value="python"
        >
          Python
        </option>
        <option
          value="java"
        >
          Java
        </option>
      </select>
      <div
        className="sc-gqjmRU elnEoP"
      >
        <a
          className="sc-jTzLTM ebxRNn"
          href="Python.driver"
          target="_blank"
        >
          <img
            alt="Driver code repository on GitHub"
            className="sc-VigVT cANOcV"
            src="github.svg"
            title="Driver code repository on GitHub"
          />
        </a>
      </div>
    </div>
    <div
      className="sc-bZQynM eJfGpj"
    >
      <label
        className="sc-EHOje dWtenm"
        htmlFor="examples-selector"
      >
        Examples
      </label>
      <select
        className="sc-htoDjs fvfSyI"
        id="examples-selector"
        onChange={[Function]}
        value={undefined}
      >
        <option
          value=""
        >
          ---
        </option>
        <option
          value="python"
        />
        <option
          value="java"
        />
      </select>
    </div>
    <div
      className="sc-bZQynM eJfGpj"
    >
      <input
        className="sc-dnqmqq acdud"
        onChange={[Function]}
        placeholder="raw gist url"
        type="url"
        value={undefined}
      />
      <button
        className="sc-iwsKbI kUlzXo"
        disabled={true}
        onClick={undefined}
      >
        load
      </button>
      <button
        className="sc-iwsKbI kUlzXo"
        disabled={true}
        onClick={[Function]}
      >
        share
      </button>
    </div>
    <div
      className="sc-gzVnrw jdktxT"
    >
      <button
        className="sc-gZMcBi kmPTEb"
        disabled={true}
        id="run-parser"
        onClick={undefined}
      >
        Run parser
      </button>
    </div>
  </div>
</header>
`;

exports[`Header has the button enabled if canParse is true 1`] = `
.dfgKLf {
  height: 70px;
  padding: 0 1rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: #f9f7f5;
  border-bottom: 1px solid #dad7d0;
  z-index: 9999;
  box-shadow: 0 5px 25px rgba(138, 128, 115, 0.14);
}

.gRdHVw {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-size: 1.5rem;
  font-weight: normal;
  padding: 0;
  margin: 0;
  height: 100%;
  border-right: 1px solid #dad7d0;
  padding-right: 1rem;
}

.iZWaVz {
  height: 40px;
}

.dtdqzi {
  margin-left: 0.8rem;
}

.gkhDQm {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-left: 1rem;
  height: 100%;
}

.dWtenm {
  color: grey;
  margin-right: 1em;
  font-size: 0.8rem;
  font-weight: bold;
  -webkit-text-transform: uppercase;
  text-transform: uppercase;
  color: #636262;
}

.eJfGpj {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 100%;
  border-right: 1px solid #dad7d0;
  padding-right: 1rem;
}

.eJfGpj + .eJfGpj {
  margin-left: 1rem;
}

.eJfGpj:last-child {
  border-right: none;
}

.jdktxT {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 100%;
  border-right: 1px solid #dad7d0;
  padding-right: 1rem;
  -webkit-flex-grow: 1;
  -ms-flex-grow: 1;
  flex-grow: 1;
  -webkit-flex-direction: row-reverse;
  -ms-flex-direction: row-reverse;
  flex-direction: row-reverse;
  padding-right: 0;
}

.jdktxT + .jdktxT {
  margin-left: 1rem;
}

.jdktxT:last-child {
  border-right: none;
}

.fvfSyI {
  border-radius: 3px;
  border: 1px solid #dad7d0;
  background: white;
  padding: 0.5em 0.5em;
  -webkit-text-transform: uppercase;
  text-transform: uppercase;
  font-weight: bold;
  font-size: 0.8rem;
  font-size: .7rem;
  -webkit-text-transform: uppercase;
  text-transform: uppercase;
}

.fvfSyI + button {
  margin-left: 3px;
}

.acdud {
  border-radius: 3px;
  border: 1px solid #dad7d0;
  background: white;
  padding: 0.5em 0.5em;
  -webkit-text-transform: uppercase;
  text-transform: uppercase;
  font-weight: bold;
  font-size: 0.8rem;
  -webkit-text-transform: none;
  text-transform: none;
}

.acdud + button {
  margin-left: 3px;
}

.kUlzXo {
  padding: 0.5em 0.5em;
  border: 1px solid #B1B1B1;
  border-radius: 3px;
  cursor: pointer;
  background: #B1B1B1;
  color: #212121;
  font-weight: bold;
  -webkit-text-transform: uppercase;
  text-transform: uppercase;
  font-size: 0.8rem;
  -webkit-letter-spacing: 0.05em;
  -moz-letter-spacing: 0.05em;
  -ms-letter-spacing: 0.05em;
  letter-spacing: 0.05em;
  -webkit-transition: box-shadow 300ms ease-in-out;
  transition: box-shadow 300ms ease-in-out;
  border: none;
}

.kUlzXo:hover {
  box-shadow: 0 5px 15px rgba(138, 128, 115, 0.8);
}

.kUlzXo + .kUlzXo {
  margin-left: 0.7rem;
}

.kmPTEb {
  padding: 0.5em 0.5em;
  border: 1px solid #B1B1B1;
  border-radius: 3px;
  cursor: pointer;
  background: #B1B1B1;
  color: #212121;
  font-weight: bold;
  -webkit-text-transform: uppercase;
  text-transform: uppercase;
  font-size: 0.8rem;
  -webkit-letter-spacing: 0.05em;
  -moz-letter-spacing: 0.05em;
  -ms-letter-spacing: 0.05em;
  letter-spacing: 0.05em;
  -webkit-transition: box-shadow 300ms ease-in-out;
  transition: box-shadow 300ms ease-in-out;
  padding: .7rem 1.8rem;
  background: #EB7422;
  color: #FFFFFF;
  white-space: nowrap;
}

.kmPTEb:hover {
  box-shadow: 0 5px 15px rgba(138, 128, 115, 0.8);
}

.kmPTEb:hover {
  box-shadow: 0 5px 15px rgba(191, 76, 50, 0.48);
}

.elnEoP {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-left: 0.5rem;
}

.cANOcV {
  width: 20px;
  opacity: 0.7;
  margin-right: 0.3rem;
}

.ebxRNn {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: black;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 0.9rem;
}

.ebxRNn:hover {
  color: #EB7422;
}

.ebxRNn:hover .sc-VigVT {
  opacity: 1;
}

<header
  className="sc-bdVaJa dfgKLf"
>
  <h1
    className="sc-bwzfXH gRdHVw"
  >
    <img
      alt="bblfsh"
      className="sc-htpNat iZWaVz"
      src="babelfish_logo.svg"
    />
    <span
      className="sc-bxivhb dtdqzi"
    >
      Dashboard
    </span>
  </h1>
  <div
    className="sc-ifAKCX gkhDQm"
  >
    <div
      className="sc-bZQynM eJfGpj"
    >
      <label
        className="sc-EHOje dWtenm"
        htmlFor="language-selector"
      >
        Language
      </label>
      <select
        className="sc-htoDjs fvfSyI"
        id="language-selector"
        onChange={[Function]}
        value=""
      >
        <option
          value=""
        >
          Python (auto)
        </option>
        <option
          value="python"
        >
          Python
        </option>
        <option
          value="java"
        >
          Java
        </option>
      </select>
      <div
        className="sc-gqjmRU elnEoP"
      >
        <a
          className="sc-jTzLTM ebxRNn"
          href="Python.driver"
          target="_blank"
        >
          <img
            alt="Driver code repository on GitHub"
            className="sc-VigVT cANOcV"
            src="github.svg"
            title="Driver code repository on GitHub"
          />
        </a>
      </div>
    </div>
    <div
      className="sc-bZQynM eJfGpj"
    >
      <label
        className="sc-EHOje dWtenm"
        htmlFor="examples-selector"
      >
        Examples
      </label>
      <select
        className="sc-htoDjs fvfSyI"
        id="examples-selector"
        onChange={[Function]}
        value={undefined}
      >
        <option
          value=""
        >
          ---
        </option>
        <option
          value="python"
        />
        <option
          value="java"
        />
      </select>
    </div>
    <div
      className="sc-bZQynM eJfGpj"
    >
      <input
        className="sc-dnqmqq acdud"
        onChange={[Function]}
        placeholder="raw gist url"
        type="url"
        value={undefined}
      />
      <button
        className="sc-iwsKbI kUlzXo"
        disabled={true}
        onClick={undefined}
      >
        load
      </button>
      <button
        className="sc-iwsKbI kUlzXo"
        disabled={true}
        onClick={[Function]}
      >
        share
      </button>
    </div>
    <div
      className="sc-gzVnrw jdktxT"
    >
      <button
        className="sc-gZMcBi kmPTEb"
        disabled={false}
        id="run-parser"
        onClick={undefined}
      >
        Run parser
      </button>
    </div>
  </div>
</header>
`;

exports[`Header renders correctly 1`] = `
.dfgKLf {
  height: 70px;
  padding: 0 1rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: #f9f7f5;
  border-bottom: 1px solid #dad7d0;
  z-index: 9999;
  box-shadow: 0 5px 25px rgba(138, 128, 115, 0.14);
}

.gRdHVw {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-size: 1.5rem;
  font-weight: normal;
  padding: 0;
  margin: 0;
  height: 100%;
  border-right: 1px solid #dad7d0;
  padding-right: 1rem;
}

.iZWaVz {
  height: 40px;
}

.dtdqzi {
  margin-left: 0.8rem;
}

.gkhDQm {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-left: 1rem;
  height: 100%;
}

.dWtenm {
  color: grey;
  margin-right: 1em;
  font-size: 0.8rem;
  font-weight: bold;
  -webkit-text-transform: uppercase;
  text-transform: uppercase;
  color: #636262;
}

.eJfGpj {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 100%;
  border-right: 1px solid #dad7d0;
  padding-right: 1rem;
}

.eJfGpj + .eJfGpj {
  margin-left: 1rem;
}

.eJfGpj:last-child {
  border-right: none;
}

.jdktxT {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 100%;
  border-right: 1px solid #dad7d0;
  padding-right: 1rem;
  -webkit-flex-grow: 1;
  -ms-flex-grow: 1;
  flex-grow: 1;
  -webkit-flex-direction: row-reverse;
  -ms-flex-direction: row-reverse;
  flex-direction: row-reverse;
  padding-right: 0;
}

.jdktxT + .jdktxT {
  margin-left: 1rem;
}

.jdktxT:last-child {
  border-right: none;
}

.fvfSyI {
  border-radius: 3px;
  border: 1px solid #dad7d0;
  background: white;
  padding: 0.5em 0.5em;
  -webkit-text-transform: uppercase;
  text-transform: uppercase;
  font-weight: bold;
  font-size: 0.8rem;
  font-size: .7rem;
  -webkit-text-transform: uppercase;
  text-transform: uppercase;
}

.fvfSyI + button {
  margin-left: 3px;
}

.acdud {
  border-radius: 3px;
  border: 1px solid #dad7d0;
  background: white;
  padding: 0.5em 0.5em;
  -webkit-text-transform: uppercase;
  text-transform: uppercase;
  font-weight: bold;
  font-size: 0.8rem;
  -webkit-text-transform: none;
  text-transform: none;
}

.acdud + button {
  margin-left: 3px;
}

.kUlzXo {
  padding: 0.5em 0.5em;
  border: 1px solid #B1B1B1;
  border-radius: 3px;
  cursor: pointer;
  background: #B1B1B1;
  color: #212121;
  font-weight: bold;
  -webkit-text-transform: uppercase;
  text-transform: uppercase;
  font-size: 0.8rem;
  -webkit-letter-spacing: 0.05em;
  -moz-letter-spacing: 0.05em;
  -ms-letter-spacing: 0.05em;
  letter-spacing: 0.05em;
  -webkit-transition: box-shadow 300ms ease-in-out;
  transition: box-shadow 300ms ease-in-out;
  border: none;
}

.kUlzXo:hover {
  box-shadow: 0 5px 15px rgba(138, 128, 115, 0.8);
}

.kUlzXo + .kUlzXo {
  margin-left: 0.7rem;
}

.kmPTEb {
  padding: 0.5em 0.5em;
  border: 1px solid #B1B1B1;
  border-radius: 3px;
  cursor: pointer;
  background: #B1B1B1;
  color: #212121;
  font-weight: bold;
  -webkit-text-transform: uppercase;
  text-transform: uppercase;
  font-size: 0.8rem;
  -webkit-letter-spacing: 0.05em;
  -moz-letter-spacing: 0.05em;
  -ms-letter-spacing: 0.05em;
  letter-spacing: 0.05em;
  -webkit-transition: box-shadow 300ms ease-in-out;
  transition: box-shadow 300ms ease-in-out;
  padding: .7rem 1.8rem;
  background: #EB7422;
  color: #FFFFFF;
  white-space: nowrap;
}

.kmPTEb:hover {
  box-shadow: 0 5px 15px rgba(138, 128, 115, 0.8);
}

.kmPTEb:hover {
  box-shadow: 0 5px 15px rgba(191, 76, 50, 0.48);
}

.elnEoP {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-left: 0.5rem;
}

.cANOcV {
  width: 20px;
  opacity: 0.7;
  margin-right: 0.3rem;
}

.ebxRNn {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: black;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 0.9rem;
}

.ebxRNn:hover {
  color: #EB7422;
}

.ebxRNn:hover .sc-VigVT {
  opacity: 1;
}

<header
  className="sc-bdVaJa dfgKLf"
>
  <h1
    className="sc-bwzfXH gRdHVw"
  >
    <img
      alt="bblfsh"
      className="sc-htpNat iZWaVz"
      src="babelfish_logo.svg"
    />
    <span
      className="sc-bxivhb dtdqzi"
    >
      Dashboard
    </span>
  </h1>
  <div
    className="sc-ifAKCX gkhDQm"
  >
    <div
      className="sc-bZQynM eJfGpj"
    >
      <label
        className="sc-EHOje dWtenm"
        htmlFor="language-selector"
      >
        Language
      </label>
      <select
        className="sc-htoDjs fvfSyI"
        id="language-selector"
        onChange={[Function]}
        value=""
      >
        <option
          value=""
        >
          Java (auto)
        </option>
        <option
          value="python"
        >
          Python
        </option>
        <option
          value="java"
        >
          Java
        </option>
      </select>
      <div
        className="sc-gqjmRU elnEoP"
      >
        <a
          className="sc-jTzLTM ebxRNn"
          href="Java.driver"
          target="_blank"
        >
          <img
            alt="Driver code repository on GitHub"
            className="sc-VigVT cANOcV"
            src="github.svg"
            title="Driver code repository on GitHub"
          />
        </a>
      </div>
    </div>
    <div
      className="sc-bZQynM eJfGpj"
    >
      <label
        className="sc-EHOje dWtenm"
        htmlFor="examples-selector"
      >
        Examples
      </label>
      <select
        className="sc-htoDjs fvfSyI"
        id="examples-selector"
        onChange={[Function]}
        value={undefined}
      >
        <option
          value=""
        >
          ---
        </option>
        <option
          value="python"
        />
        <option
          value="java"
        />
      </select>
    </div>
    <div
      className="sc-bZQynM eJfGpj"
    >
      <input
        className="sc-dnqmqq acdud"
        onChange={[Function]}
        placeholder="raw gist url"
        type="url"
        value={undefined}
      />
      <button
        className="sc-iwsKbI kUlzXo"
        disabled={true}
        onClick={undefined}
      >
        load
      </button>
      <button
        className="sc-iwsKbI kUlzXo"
        disabled={true}
        onClick={[Function]}
      >
        share
      </button>
    </div>
    <div
      className="sc-gzVnrw jdktxT"
    >
      <button
        className="sc-gZMcBi kmPTEb"
        disabled={true}
        id="run-parser"
        onClick={undefined}
      >
        Run parser
      </button>
    </div>
  </div>
</header>
`;
